src/transaction/log_page_buffer.c

```diff
@@ -3361,6 +3364,250 @@ logpb_write_toflush_pages_to_archive (THREAD_ENTRY * thread_p)
+
+	// The temporary EOF log record at the prev_lsa has to be removed.
+	log_Pb.partial_append.status = LOGPB_APPENDREC_PARTIAL_ENDED;
+	logpb_flush_all_append_pages (thread_p);
+	assert (log_Pb.partial_append.status == LOGPB_APPENDREC_PARTIAL_FLUSHED_ORIGINAL);
+
+	// A new EOF has been appended at the append_lsa in logpb_flush_all_append_pages. Remove it as well.
+	LOG_RECORD_HEADER *append_log_rec = LOG_GET_LOG_RECORD_HEADER (log_Gl.append.log_pgptr, &log_Gl.hdr.append_lsa);
+	assert (append_log_rec->type == LOG_END_OF_LOG);
+	assert (log_Gl.append.log_pgptr->hdr.logical_pageid == log_pgptr->hdr.logical_pageid);
+	memcpy (append_log_rec, origin_append_log_record, sizeof (LOG_RECORD_HEADER));
+
+	logpb_set_dirty (thread_p, log_Gl.append.log_pgptr);
+      }
+      break;
+    default:
+      /* invalid state */
+      assert_release (false);
+    }
+  log_Pb.partial_append.status = LOGPB_APPENDREC_SUCCESS;
+}
+
+/*
+ * logpb_catchup_finish - finish the catch-up job. 
+ *      This sets the prev_lsa and append_lsa to the latest ones traversing log pages from known log record.
+ *
+ * return: NO_ERROR
+ *
+ *   catchup_lsa(in): catchup_lsa, until which we pull pages, and from which new log records will be appended.
+ */
+int
+logpb_catchup_finish (THREAD_ENTRY * thread_p, const LOG_LSA catchup_lsa)
+{
+  assert (LOG_CS_OWN_WRITE_MODE (thread_p));
+  assert (log_Pb.partial_append.status = LOGPB_APPENDREC_SUCCESS);
+
+  char log_pgbuf[LOG_PAGESIZE + MAX_ALIGNMENT], *aligned_log_pgbuf;
+  LOG_LSA prev_lsa = log_Gl.append.prev_lsa;
+  LOG_LSA nav_lsa = log_Gl.append.prev_lsa;
+  LOG_PAGE *log_pgptr = NULL;
+  int error = NO_ERROR;
+
+  aligned_log_pgbuf = PTR_ALIGN (log_pgbuf, MAX_ALIGNMENT);
+  log_pgptr = (LOG_PAGE *) aligned_log_pgbuf;
+
+  error = logpb_fetch_page (thread_p, &prev_lsa, LOG_CS_FORCE_USE, log_pgptr);
+  if (error != NO_ERROR)
+    {
+      return error;
+    }
+
+  while (nav_lsa < catchup_lsa)
+    {
+      if (nav_lsa.pageid != prev_lsa.pageid)
+	{
+	  error = logpb_fetch_page (thread_p, &nav_lsa, LOG_CS_FORCE_USE, log_pgptr);
+	  if (error != NO_ERROR)
+	    {
+	      return error;
+	    }
+	}
+
+      // If offset is missing, it is because we archive an incomplete log record.
+      if (nav_lsa.offset == NULL_OFFSET)
+	{
+	  nav_lsa.offset = log_pgptr->hdr.offset;
+	  if (nav_lsa.offset == NULL_OFFSET)
+	    {
+	      // There is nothing in this page. Here is a part of incomplete log record.
+	      nav_lsa.pageid++;
+	      continue;
+	    }
+	}
+
+      prev_lsa = nav_lsa;
+
+      LOG_RECORD_HEADER *log_rec = LOG_GET_LOG_RECORD_HEADER (log_pgptr, &prev_lsa);
+
+      nav_lsa = log_rec->forw_lsa;
+
+      if (nav_lsa.is_null () && logpb_is_page_in_archive (prev_lsa.pageid))
+	{
+	  nav_lsa.pageid = prev_lsa.pageid + 1;
+	  nav_lsa.offset = NULL_OFFSET;
+	}
+    }
+
+  assert_release (nav_lsa == catchup_lsa);
+
+  log_Gl.append.prev_lsa = prev_lsa;
+  log_Gl.hdr.append_lsa = catchup_lsa;
```

Comment:
At the recovery phase, `eof_lsa` may be used rarely, but it is also used in emergency mode (log_manager.c:1426, log_initialize_internal()). Therefore, I think it would be good to update it with the accurate `append_lsa` value. Or, could it be updated elsewhere?
```suggestion
  log_Gl.hdr.append_lsa = catchup_lsa;
  log_Gl.hdr.eof_lsa = log_Gl.hdr.append_lsa;
```

Reply:
Could you tell me why we should keep the `eof_lsa` updated to the `append_lsa` here? What I understand is that `eof_lsa` is the location where the EOF log record is appended while flushing log pages. This will be updated while flushing, and here the `append_lsa` doesn't points to the EOL log record.

Reply:
I think I briefly got confused about this. I'm sorry